package/MDAnalysis/analysis/nuclinfo.py

```diff
@@ -763,3 +771,118 @@ def pseudo_dihe_baseflip(universe, bp1, bp2, i,
+       doi: `10.1021/jp209986y`_
+
+    .. _`10.1021/jp209986y`: https://doi.org/10.1021/jp209986y
+
+    .. Note:: This angle calculation will only work if using atom names as
+              documented by charmm force field parameters.
+
+    Parameters
+    ----------
+    universe : Universe
+        :class:`~MDAnalysis.core.universe.Universe` containing the
+        trajectory
+    b1 : int
+        resid of the first base
+    b2 : int
+        resid of the second base
+    segid1 : str (optional)
+        segid of b1
+    segid2 : str (optional)
+        segid of b2
+    box : numpy.ndarray (optional)
+        The unitcell dimensions of the system, which can be orthogonal or
+        triclinic and must be provided in the same format as returned by
+        :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:
+        ``[lx, ly, lz, alpha, beta, gamma]``.    
+
+    Returns
+    -------
+    float
+          angle in degrees
+
+
+    .. versionadded:: 2.1.0
+    """
+    batms = {'CYT': ['N1', 'N3', 'C5'],
+             'THY': ['N1', 'N3', 'C5'],
+             'URA': ['N1', 'N3', 'C5'],
+             'ADE': ['N9', 'N1', 'N7'],
+             'GUA': ['N9', 'N1', 'N7'],
+             'C': ['N1', 'N3', 'C5'],
+             'T': ['N1', 'N3', 'C5'],
+             'U': ['N1', 'N3', 'C5'],
+             'A': ['N9', 'N1', 'N7'],
+             'G': ['N9', 'N1', 'N7']}
+    # select residues
+    bf1 = universe.select_atoms(
+        "(segid {0!s} and resid {1!s})".format(seg1, b1))
+    bf2 = universe.select_atoms(
+        "(segid {0!s} and resid {1!s})".format(seg2, b2))
+    for ag, segid, res in [(bf1, seg1, b1), (bf2, seg2, b2)]:
+        if len(ag.residues) > 1:
+            warnings.warn(f"Found more than one residues with resid {res} and "
+                          f"segid {segid}. Using first residue in selection.")
+    # get residue names
+    resn1 = bf1.atoms.resnames[0]
+    resn2 = bf2.atoms.resnames[0]
+    # extract positions for specific atoms of each base
+    try:
+        c11 = bf1.select_atoms("name {0!s}".format(batms[resn1][0])).positions[0]
+        c12 = bf1.select_atoms("name {0!s}".format(batms[resn1][1])).positions[0]
+        c13 = bf1.select_atoms("name {0!s}".format(batms[resn1][2])).positions[0]
+        c21 = bf2.select_atoms("name {0!s}".format(batms[resn2][0])).positions[0]
+        c22 = bf2.select_atoms("name {0!s}".format(batms[resn2][1])).positions[0]
+        c23 = bf2.select_atoms("name {0!s}".format(batms[resn2][2])).positions[0]
+    except KeyError:
+        errmsg = ("Residue names may be incorrect! "
+                  "Please use either ADE/GUA/CYT/URA/THY or A/G/C/U/T format.")
+        raise KeyError(errmsg)
+    except IndexError:
+        errmsg = ("Atom names may be incorrect! "
+                  "Make sure base atom names follow CHARMM format.")
+        raise IndexError(errmsg)
+    # check box
+    if box is None:
+        box = universe.dimensions
+        if box is None:
```

Comment:
Don't invent a box if there isn't one, it's incredibly hard to guess a box. Better to only do the call to minimize if there's a box, otherwise trust the vectors to be correct.